pkg/disttask/framework/storage/task_table.go

```diff
@@ -555,6 +555,26 @@ func (mgr *TaskManager) UpdateErrorToSubtask(ctx context.Context, execID string,
 	return err1
 }
 
+// CancelSubtask update the task's subtasks' state to canceled.
+func (mgr *TaskManager) CancelSubtask(ctx context.Context, execID string, taskID int64) error {
+	_, err1 := mgr.ExecuteSQLWithNewSession(ctx,
+		`update mysql.tidb_background_subtask
+		set state = %?, 
+		start_time = unix_timestamp(), 
+		state_update_time = unix_timestamp(),
+		end_time = CURRENT_TIMESTAMP()
+		where exec_id = %? and 
+		task_key = %? and 
+		state in (%?, %?) 
+		limit 1;`,
+		proto.SubtaskStateCanceled,
+		execID,
+		taskID,
+		proto.TaskStatePending,
+		proto.TaskStateRunning)
```

Comment:
```suggestion
		proto.SubtaskStatePending,
		proto.SubtaskStateRunning)
```

Reply:
https://github.com/pingcap/tidb/pull/50429/commits/015ac40079ffa1e33a5535a204107ffef3b68d2d
So many place need to change, will fire another pr to change all of them since this pr needs to cp

Reply:
lgtm